code/iaas/model/src/main/java/io/cattle/platform/core/dao/impl/ClusterHostMapDaoImpl.java

```diff
@@ -41,4 +47,26 @@ public ClusterHostMapRecord getClusterHostMap(Host cluster, Host host) {
                 .fetchOneInto(ClusterHostMapRecord.class);
     }
 
+    @Override
+    public IpAddress getIpAddressForHost(Long hostId) {
+        return create()
+            .select(IP_ADDRESS.fields())
+                .from(IP_ADDRESS)
+                .join(HOST_IP_ADDRESS_MAP)
+                    .on(IP_ADDRESS.ID.eq(HOST_IP_ADDRESS_MAP.IP_ADDRESS_ID))
+                .where(HOST_IP_ADDRESS_MAP.HOST_ID.eq(hostId)
+                    .and(HOST_IP_ADDRESS_MAP.REMOVED.isNull())
+                    .and(IP_ADDRESS.REMOVED.isNull()))
+            .fetchOneInto(IpAddressRecord.class);
+    }
+
+    @Override
+    public Host findHostByName(String name) {
```

Comment:
Looks like cattle allows duplicated names for the host, so this record might result in fetching just one record out of many. If there is a backend restriction on the hostName to be unique within the cluster, we define this restriction on the cattle side as well.